src/main/java/executor/service/service/scenarios/impl/JSONFileScenarioProvider.java

```diff
@@ -0,0 +1,73 @@
+package executor.service.service.scenarios.impl;
+
+import com.google.common.reflect.TypeToken;
+import com.google.gson.*;
+import executor.service.model.scenario.Scenario;
+import executor.service.model.scenario.StepAction;
+import executor.service.service.scenarios.ScenarioProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.util.List;
+
+/**
+ * The {@code JSONFileScenarioProvider} class is an implementation of the {@link ScenarioProvider}
+ * interface. It is responsible for reading scenarios from a JSON file and providing them as a list
+ * of {@link Scenario} objects.
+ * @author Yurii Kotsiuba
+ * @see ScenarioProvider
+ * @see Scenario
+ */
+public class JSONFileScenarioProvider implements ScenarioProvider {
+
+    private static final Logger log = LoggerFactory.getLogger(ScenarioProvider.class);
+    private static final String FILE_NAME = "/scenarios.json";
```

Comment:
The file name is now hardcoded as **/scenarios.json**. This may not be practical if you need to rename the file. You may want to consider passing the filename to the JSONFileScenarioProvider constructor or using a configuration file to store the filename.

Reply:
From what place do you read this files?